cloudsql/mysql/database-sql/cloudsql.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample database-sql demonstrates connection to a Cloud SQL instance from App Engine
+// standard. The application is a Golang version of the "Tabs vs Spaces" web
+// app presented at Cloud Next '19 as seen in this video:
+// https://www.youtube.com/watch?v=qVgzP3PsXFw&t=1833s
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"html/template"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/go-sql-driver/mysql"
+)
+
+// db is the global database connection pool.
+var db *sql.DB
+
+// parsedTemplate is the global parsed HTML template.
+var parsedTemplate *template.Template
+
+// vote struct contains a single row from the votes table in the database.
+// Each vote includes a candidate ("TABS" or "SPACES") and a timestamp.
+type vote struct {
+	Candidate string
+	VoteTime  mysql.NullTime
+}
+
+// voteDiff is used to provide a string representation of the current voting
+// margin, such as "1 vote" (singular) or "2 votes" (plural).
+type voteDiff int
+
+func (v voteDiff) String() string {
+	if v == 1 {
+		return "1 vote"
+	}
+	return strconv.Itoa(int(v)) + " votes"
+}
+
+// templateData struct is used to pass data to the HTML template.
+type templateData struct {
+	TabsCount   uint
+	SpacesCount uint
+	VoteMargin  string
+	RecentVotes []vote
+}
+
+func main() {
+	var err error
+
+	parsedTemplate, err = template.ParseFiles("templates/index.html")
+	if err != nil {
+		log.Fatalf("unable to parse template file: %s", err)
+	}
+
+	// If the optional DB_TCP_HOST environment variable is set, it contains
+	// the IP address and port number of a TCP connection pool to be created,
+	// such as "127.0.0.1:3306". If DB_TCP_HOST is not set, a Unix socket
+	// connection pool will be created instead.
+	if os.Getenv("DB_TCP_HOST") == "" {
```

Comment:
nit: I would reverse this to match the comment. Right now it's actually checking if var is unset (or empty) use socket, otherwise use TCP